mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt

```diff
@@ -1276,16 +1276,20 @@ class WidgetAdapter(
 
             // hide dividers before and after frame widgets
             val adapter = parent.adapter
-            val noDividerTypes = intArrayOf(TYPE_FRAME, TYPE_INVISIBLE)
             if (adapter != null) {
-                if (adapter.getItemViewType(position) in noDividerTypes) {
+                if (adapter.getItemViewType(position) == TYPE_FRAME) {
                     return true
                 }
                 if (position < adapter.itemCount - 1) {
-                    if (adapter.getItemViewType(position + 1) in noDividerTypes) {
+                    if (adapter.getItemViewType(position + 1) in intArrayOf(TYPE_FRAME, TYPE_INVISIBLE)) {
                         return true
                     }
                 }
+                if (adapter.getItemViewType(position) == TYPE_INVISIBLE &&
+                    adapter.getItemViewType(position + 1) == TYPE_INVISIBLE
```

Comment:
This needs to be in the if check above, to not run into an AIOOBE for the last item in the list, where `position + 1` exceeds the valid list indices.